tests/test_iconify.py

```diff
@@ -0,0 +1,22 @@
+from typing import TYPE_CHECKING
+
+import pytest
+from qtpy.QtWidgets import QPushButton
+
+from superqt import QIconifyIcon
+
+if TYPE_CHECKING:
+    from pytestqt.qtbot import QtBot
+
+
+def test_qiconify(qtbot: "QtBot", monkeypatch: "pytest.MonkeyPatch") -> None:
+    monkeypatch.setenv("PYCONIFY_CACHE", "0")
```

Comment:
```suggestion
def test_qiconify(qtbot: "QtBot", monkeypatch: "pytest.MonkeyPatch", tmp_path: "pathlib.Path") -> None:
    monkeypatch.setenv("PYCONIFY_CACHE", str(tmp_path))
```

Instead of depending on pyconify clean mechanism just store cache in test managed directory?